app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class CartsController < ApplicationController
+  rescue_from ActiveRecord::RecordNotFound, with: :invalid_cart
+  before_action :set_cart, only: %i[show destroy]
+  # rubocop:disable Rails/I18nLocaleTexts
+
+  # GET /carts/1
+  # GET /carts/1.json
+  def show; end
+
+  # GET /carts/new
+  def new
+    @cart = Cart.new
+  end
+
+  # POST /carts
+  # POST /carts.json
+  def create
+    @cart = Cart.new(cart_params)
+
+    respond_to do |format|
+      if @cart.save
+        format.html { redirect_to @cart, notice: 'Cart was successfully created.' }
+        format.json { render :show, status: :created, location: @cart }
+      else
+        format.html { render :new }
+        format.json { render json: @cart.errors, status: :unprocessable_entity }
+      end
+    end
+  end
```

Comment:
You are setting a cart in `ApplicationController` -> `before_action :set_cart` (which is valid throughout the whole app). I don't see why you have new and create actions here. Can I get more context please?